code/modules/mining/drilling/drill.dm

```diff
@@ -189,8 +191,46 @@
 		else
 			user << "<span class='notice'>The drill is unpowered.</span>"
 	else
-		user << "<span class='notice'>Turning on a piece of industrial machinery without sufficient bracing or wires exposed is a bad idea.</span>"
-
+		if(use_cell_power())
+			if(!supported && !panel_open)
+				system_error("unbraced drill error")
+				sleep(30)
+				if(!supported) //if you can resolve it manually in three seconds then power to you good-sir.
+					if(prob(50))
+						visible_message("<span class='notice'>\icon[src] [src.name] beeps, \"Unbraced drill error automatically corrected. Please brace your drill.\"</span>")
+					else
+						visible_message("<span class='danger'>\The [src] explodes!</span>")
+						fragem(src,10,35,2,1,5,1,0)
+						qdel(src)
+				else
+					visible_message("<span class='notice'>\icon[src] [src.name] beeps, \"Unbraced drill error manually resolved. Operations may resume normally.\"</span>")
+			if(supported && panel_open)
+				if(cell)
+					system_error("unsealed cell fitting error")
+					var/datum/effect/effect/system/spark_spread/s = new /datum/effect/effect/system/spark_spread
+					s.set_up(3, 1, src.loc)
+					s.start()
+					sleep(20)
+					s.set_up(3, 1, src.loc)
+					s.start()
+					sleep(10)
+					s.set_up(3, 1, src.loc)
+					s.start()
+					sleep(10)
+					if(panel_open)
+						if(prob(70))
+							visible_message("<span class='danger'>\The [src]'s cell shorts out!</span>")
+							cell.use(cell.charge)
+						else
+							visible_message("<span class='danger'>\The [src]'s cell detonates!</span>")
+							explosion(src.loc, -1, -1, 2, 1)
+							qdel(cell)
+							cell = null
```

Comment:
Removing null from component_parts is not really useful. I might have been a little incoherent when correcting you last night, as I was 2 seconds from passing out.

```
qdel(cell)
component_parts -= cell
cell = null
```

This is the proper order of operations.